src/java/org/apache/cassandra/service/reads/AbstractReadExecutor.java

```diff
@@ -201,6 +208,15 @@ public static AbstractReadExecutor getReadExecutor(SinglePartitionReadCommand co
 
         ReplicaPlan.ForTokenRead replicaPlan = ReplicaPlans.forRead(keyspace, command.partitionKey().getToken(), command.indexQueryPlan(), consistencyLevel, retry);
 
+        if (replicaPlan.candidates().stream().noneMatch(replica -> replica.endpoint().equals(FBUtilities.getBroadcastAddressAndPort())))
+        {
+            ReadCoordinationMetrics.nonreplicaRequests.inc();
+        }
+        else if (replicaPlan.contacts().stream().noneMatch(replica -> replica.endpoint().equals(FBUtilities.getBroadcastAddressAndPort())))
+        {
+            ReadCoordinationMetrics.preferredOtherReplicas.inc();
+        }
+
```

Comment:
It wasn't quite clear to me, but in this port I translated `scratchAllReplicas` -> `replicaPlan.candidates` and `scratchTargetReplicas` to `replicaPlan.contacts`.

Reply:
I wasn't able to identify how `contacts` and `candidates` populated. But it seems to me that your conclusion is correct.